ui.apps/src/main/javascript/minicart/src/components/Minicart/minicart.js

```diff
@@ -44,53 +41,39 @@ const MiniCart = props => {
         refetchQueries: [{ query: CART_DETAILS_QUERY, variables: { cartId } }]
     });
 
-    const { data, error, loading: queryLoading } = useQuery(CART_DETAILS_QUERY, { variables: { cartId } });
+    const { data, error, loading: queryLoading } = useQuery(CART_DETAILS_QUERY, {
+        variables: { cartId },
+        skip: !cartId
+    });
+
+    useEffect(() => {
+        if (data && data.cart) {
+            dispatch({ type: 'cart', cart: data.cart });
+        }
+    }, [data]);
 
     if (error) {
         console.error(`Error loading cart`, error);
     }
 
-    const openCart = () => {
-        setIsOpen(true);
-    };
-
     const addToCart = ev => {
         if (!ev.detail) {
             return;
         }
 
         const { sku, quantity } = ev.detail;
         addItem({ variables: { cartId, sku, quantity } });
-        setIsOpen(true);
+        dispatch({ type: 'open' });
     };
 
-    useEventListener(document, 'aem.cif.open-cart', openCart);
-    useEventListener(document, 'aem.cif.add-to-cart', addToCart);
-
-    const handleCloseCart = useCallback(() => {
-        setIsOpen(false);
-    });
-
-    const handleResetCart = useCallback(() => {
-        resetCart();
-        handleCloseCart();
-    });
-
-    const handleBeginEditing = useCallback(item => {
-        setIsEditing(true);
-        setEditItem(item);
-    });
-
-    const handleEndEditing = useCallback(() => {
-        setIsEditing(false);
+    useEventListener(document, 'aem.cif.open-cart', () => {
+        dispatch({ type: 'open' });
     });
+    useEventListener(document, 'aem.cif.add-to-cart', addToCart);
 
-    const removeItemFromCart = useCallback(
-        itemId => {
-            removeItem({ variables: { cartId, itemId } });
-        },
-        [removeItem]
-    );
+    const removeItemFromCart = itemId => {
```

Comment:
This handler should be provided by the context.

Reply:
I updated the code to make those two methods available via the context. Please also check the update in the PR description.